diesel/src/query_dsl/save_changes_dsl.rs

```diff
@@ -66,45 +66,46 @@ where
     }
 }
 
+/// Sugar for types which implement both `AsChangeset` and `Identifiable`
+///
+/// `foo.save_changes(&conn)` is equivalent to
+/// `update(foo::table().find(foo.id())).set(&foo).get_result(&conn)`
+///
+/// # Example
+///
+/// ```rust
+/// # #[macro_use] extern crate diesel;
+/// # include!("../doctest_setup.rs");
+/// # use schema::animals;
+/// #
+/// # #[derive(AsChangeset, Debug, PartialEq, Identifiable, Queryable)]
+/// # struct Animal {
+/// #    id: i32,
+/// #    species: String,
+/// #    legs: i32,
+/// #    name: Option<String>,
+/// # }
+/// #
+/// # fn main() {
+/// #     use animals::dsl::*;
+/// #     let connection = establish_connection();
+/// let mut spider = animals.filter(species.eq("spider"))
+///     .first::<Animal>(&connection)
+///     .expect("Too scary to load");
```

Comment:
maybe change this test to use a wrapper fn and `?` too?

Reply:
This is just moved from its original location, and I also have a PR up that rewrites this example